jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ConcurrentPool.java

```diff
@@ -597,4 +667,100 @@ public String toString()
                 getPooled());
         }
     }
+
+    public interface Holder<P>
+    {
+        Pool.Entry<P> getEntry();
+
+        void hold();
+
+        void free();
+    }
+
+    private static class StrongHolder<P> implements Holder<P>
+    {
+        private final ConcurrentEntry<P> _strong;
+
+        StrongHolder(ConcurrentEntry<P> entry)
+        {
+            _strong = entry;
+        }
+
+        @Override
+        public Entry<P> getEntry()
+        {
+            return _strong;
+        }
+
+        @Override
+        public void hold()
+        {
+        }
+
+        @Override
+        public void free()
+        {
+        }
+
+        @Override
+        public String toString()
+        {
+            return "%s@%x{%s}".formatted(this.getClass().getSimpleName(), hashCode(), _strong);
+        }
+    }
+
+    private static class WeakHolder<P> implements Holder<P>
+    {
+        private final WeakReference<ConcurrentEntry<P>> _weak;
+        private volatile ConcurrentEntry<P> _strong;
+
+        WeakHolder(ConcurrentEntry<P> entry)
+        {
+            _weak = new WeakReference<>(entry);
+        }
+
+        @Override
+        public Entry<P> getEntry()
+        {
+            return _weak.get();
+        }
+
+        @Override
+        public void hold()
+        {
+            ConcurrentEntry<P> entry = _weak.get();
+            if (entry == null)
+            {
+                LOG.warn("LEAKED {}", this);
```

Comment:
These should be removed, because if it happens it will print millions of warnings in logs.
Plus no information about the leak, so the log line would be kinda useless.
Yes we leak but we do not accumulate them.
If we want to discover leaks, they are better tracked at the higher level (e.g. ABBP.Tracking).

Reply:
I'll call a protected method that will do a debug for now, but can be extended if necessary.

Reply:
I've added debug mechanism that reports more info on a leaked entry.  Thoughts?